### DISCUSSION TITLE
Cannot use different slug names for the same dynamic path

### DISCUSSION THREAD

**Discussion by pateljoel** [vercel/next.js#14393] [Help]
I have a Next.js app and would like to create multiple dynamically generated pages at the **root** of the URL (no nesting). 

Example:

```
example.com/[topic]
example.com/[collection]
```

Both of these files are in separate js files in the pages folder (`[topic].js`, `[collection].js`) like this:

```
├── pages
│   ├── [topic].js
│   ├── [collection].js
│   ├── index.js
```

After I run my app, I get this error:
_(You cannot use different slug names for the same dynamic path ('topic' !== 'collection'))_

Is there another way around this error? i've looked everywhere for a solution to no avail.
Thanks.

**Comment by revskill10** [comment]
What's the difference between `topic_1` and `topic_2` ?

**Reply by pateljoel** [reply]
@revskill10 I changed the example above to reflect a collection or topic, I'm building a Q/A site.

The point is, I would like both pages to be generated at the root / the same level.

**Reply by revskill10** [reply]
OK, here's how i do it in my admin dashboard.

`pages/admin/[mod].js`

```js
import React from 'react'
import dynamic from 'next/dynamic'
const Cms =  dynamic(() => import('src/components/admin/cms'), { ssr: false })
const Gallery =  dynamic(() => import('src/components/admin/gallery'), { ssr: false })
const Index = ({ params }) => {
  const { mod } = params
  return (
    <Choose>
      <When condition={ mod === 'cms' }>
        <Cms />   
      </When>
      <When condition={ mod === 'gallery' }>
        <Gallery />   
      </When>
      <Otherwise>
        <span>Not found</span>
      </Otherwise>
    </Choose>
  )
}
export default Index
export const getStaticProps = async ({ params }) => {
  return {
    props: { params }
  }
}
export const getStaticPaths = () => {
  const paths = [
    { params: { mod: 'cms' }},
    { params: { mod: 'gallery' }}
  ]
  return {
    paths,
    fallback: false,
  }
}
```
Hope this helps.

**Comment by pateljoel** [comment]
I should also mention that I am using `getStaticPaths` and `getStaticProps` to generate the pages.

**Comment by IamLizu** [comment]
Okay, if you are generating pages dynamically, why two files? (`[topic].js, [collection].js`)

Create a dynamic page, maybe `[page].js`, pass your page information to this file to create pages dynamically. 

I did something similar, I generated pages dynamically from markdown files from a directory.

**Reply by pateljoel** [reply]
@IamLizu Is there an example/example code on how to do this? I've had a look at the [next.js example repository](https://github.com/vercel/next.js/tree/canary/examples) but don't see anything that resembles my usecase.

**Reply by IamLizu** [reply]
Yes, here's an example of what I said earlier,

Request page like this, see full example here: https://github.com/IamLizu/humous/blob/master/components/layout.jsx
```
<Link href="/[page]" as={page} passHref>
    <a>{title}</a>
</Link>
```

Dynamic page generation file: https://github.com/IamLizu/humous/blob/master/pages/%5Bpage%5D.jsx
